[ML] Out-of-bounds access in CEventRateModelTest/testCorrelatedNoTrend and testCorrelatedTrend

After upgrading to Boost 1.77 the Boost code contains more assertions and is revealing an out-of-bounds access in a couple of tests.

From https://elasticsearch-ci.elastic.co/job/elastic+machine-learning+main+windows+debug/97/consoleText:

```
2021-11-04 07:23:50,624850 UTC [5360] DEBUG CTestObserver.cc@28 +---------------------------------------------+

2021-11-04 07:23:50,624850 UTC [5360] DEBUG CTestObserver.cc@29 |  CEventRateModelTest/testCorrelatedNoTrend  |

2021-11-04 07:23:50,624850 UTC [5360] DEBUG CTestObserver.cc@30 +---------------------------------------------+
2021-11-04 07:23:50,625841 UTC [5360] DEBUG CEventRateModelTest.cc@657 Test correlation anomalies

Assertion failed: this->m_holder.m_size > n, file C:\usr\local\include\boost-1_77\boost/container/vector.hpp, line 1651
```

From https://elasticsearch-ci.elastic.co/job/elastic+machine-learning+main+macos-aarch64+debug/98/consoleText:

```
2021-11-04 05:28:18,666655 UTC [96482] DEBUG CTestObserver.cc@28 +---------------------------------------------+
2021-11-04 05:28:18,666663 UTC [96482] DEBUG CTestObserver.cc@29 |  CEventRateModelTest/testCorrelatedNoTrend  |
2021-11-04 05:28:18,666669 UTC [96482] DEBUG CTestObserver.cc@30 +---------------------------------------------+
2021-11-04 05:28:18,667235 UTC [96482] DEBUG CEventRateModelTest.cc@657 Test correlation anomalies
unknown location:0: fatal error: in "CEventRateModelTest/testCorrelatedNoTrend": signal: SIGABRT (application abort requested)
Assertion failed: (this->m_holder.m_size > n), function operator[], file /usr/local/include/boost-1_77/boost/container/vector.hpp, line 1651.
2021-11-04 05:28:19,153827 UTC [96482] INFO  CTestObserver.cc@40 Unit test timing - CEventRateModelTest/testCorrelatedNoTrend took 487ms
CEventRateModelTest.cc:691: last checkpoint
2021-11-04 05:28:19,153882 UTC [96482] DEBUG CTestObserver.cc@28 +-------------------------------------------+
2021-11-04 05:28:19,153892 UTC [96482] DEBUG CTestObserver.cc@29 |  CEventRateModelTest/testCorrelatedTrend  |
2021-11-04 05:28:19,153899 UTC [96482] DEBUG CTestObserver.cc@30 +-------------------------------------------+
unknown location:0: fatal error: in "CEventRateModelTest/testCorrelatedTrend": signal: SIGABRT (application abort requested)
CEventRateModelTest.cc:894: last checkpoint
```

From https://elasticsearch-ci.elastic.co/job/elastic+machine-learning+main+linux-aarch64+debug/98/consoleText:

```
2021-11-04 06:48:06,940414 UTC [5775] DEBUG CTestObserver.cc@28 +---------------------------------------------+
2021-11-04 06:48:06,940434 UTC [5775] DEBUG CTestObserver.cc@29 |  CEventRateModelTest/testCorrelatedNoTrend  |
2021-11-04 06:48:06,940449 UTC [5775] DEBUG CTestObserver.cc@30 +---------------------------------------------+
2021-11-04 06:48:06,941462 UTC [5775] DEBUG CEventRateModelTest.cc@657 Test correlation anomalies
ml_test: /usr/local/gcc103/include/boost-1_77/boost/container/vector.hpp:1651: boost::container::vector<T, Allocator, Options>::reference boost::container::vector<T, Allocator, Options>::operator[](boost::container::vector<T, Allocator, Options>::size_type) [with T = long unsigned int; A = boost::container::small_vector_allocator<long unsigned int, boost::container::new_allocator<void>, void>; Options = void; boost::container::vector<T, Allocator, Options>::reference = long unsigned int&; boost::container::vector<T, Allocator, Options>::size_type = long unsigned int]: Assertion `this->m_holder.m_size > n' failed.
si_signo 6, si_code: -6, si_errno: 0, address: 0xffffa4875238, library: /lib64/libc.so.6, base: 0xffffa4840000, normalized address: 0x35238
make[3]: *** [test] Aborted
```

--- a/lib/maths/time_series/CTimeSeriesModel.cc
+++ b/lib/maths/time_series/CTimeSeriesModel.cc
@@ -2124,8 +2124,8 @@ bool CTimeSeriesCorrelations::correlationModels(std::size_t id,
             continue;
         }
         correlated[end] = correlate;
-        variables.push_back(std::move(variable));
         correlationModels.push_back({i->second.first.get(), variable[0]});
+        variables.push_back(std::move(variable));
         ++end;
     }
 

